lib/babel/transformation/transformers/es6/parameters.rest.js

```diff
@@ -19,6 +77,24 @@ exports.Function = function (node, parent, scope) {
   // otherwise `arguments` will be remapped in arrow functions
   argsId._ignoreAliasFunctions = true;
 
+  // check if rest is used only in member expressions
+  var restOuterDeclar = scope.getBindingIdentifier(rest.name);
+  var state = {
+    name: rest.name,
+    outerDeclar: restOuterDeclar,
+    isOptimizable: true,
+    candidates: []
+  };
+  scope.traverse(node, memberExpressionVisitor, state);
+
+  if (state.isOptimizable) {
```

Comment:
if (state.isOptimizable) check is required, because I have to be abslutely sure, that every reference to rest is optimizable. If this was done visitor, it could lead to a situation, where some references are updated and some are not. It should never lead to broken code, however it look very strange.